packages/components/src/components/input-field/input-field.tsx

```diff
@@ -58,18 +119,17 @@ const InputField = ({
     type,
     unit,
     value,
-    data_testid,
-}) => {
-    const [local_value, setLocalValue] = React.useState();
-    const Icon = icon;
+}: TInputFieldProps) => {
+    const [local_value, setLocalValue] = React.useState('');
+    const Icon = icon as React.ElementType;
     const has_error = error_messages && !!error_messages.length;
     const max_is_disabled = max_value && (+value >= +max_value || +local_value >= +max_value);
     const min_is_disabled = min_value && (+value <= +min_value || +local_value <= +min_value);
     let has_valid_length = true;
 
-    const changeValue = (e, callback) => {
+    const changeValue = (e: TChangeEvent, callback?: (evt: TChangeEvent) => void) => {
         if (unit) {
-            e.target.value = e.target.value.replace(unit, '').trim();
+            e.target.value = e.target.value.toString().replace(unit, '').trim();
```

Comment:
```suggestion
            e.target.value = e.target.value.replace(unit, '').trim();
```
@jim-deriv will work if https://github.com/niloofar-deriv/deriv-app/pull/83/files#r1118250922 is applied)